src/borrowck/6.md

```diff
@@ -0,0 +1,22 @@
+# Borrowck 6 @adotinthevoid @Nilstrieb
+
+{{#include ../include/quiz-is-wip.md}}
+
+```rust
+{{#include ../../code/examples/borrowck_6.rs}}
+```
+
+<details>
+<summary>Solution</summary>
+
+```
+{{#include ../../code/examples/stderr/borrowck_6.stderr}}
+```
+
+`r2_twice` does not compile because `R2` does not implement `Copy`, and is moved twice.
+This is basic Rust, but `r1_twice` *does* compile even though `&mut i32` does *not* implement `Copy`!
+That's because for `R1`, the reference is not moved into `r1_once`, it is instead *reborrowed*.
+Reborrows are a special operation on references that the borrow checker understands and behave like `&mut *x`.
+So there is just a new reference passed to `r1_once`, with the original one preserved.
```

Comment:
Good explaination, thanks for writing it.

Reply:
:3 thansk